tests/ui/or-patterns/exhaustiveness-unreachable-pattern.stderr

```diff
@@ -13,212 +15,310 @@ LL | #![deny(unreachable_patterns)]
 error: unreachable pattern
   --> $DIR/exhaustiveness-unreachable-pattern.rs:13:9
    |
+LL |         (1 | 2,) => {}
+   |         -------- matches all the values already
 LL |         (2,) => {}
-   |         ^^^^
+   |         ^^^^ unreachable pattern
 
 error: unreachable pattern
   --> $DIR/exhaustiveness-unreachable-pattern.rs:19:9
    |
+LL |         (1,) => {}
+   |         ---- matches some of the same values
+LL |         (2,) => {}
+   |         ---- matches some of the same values
 LL |         (1 | 2,) => {}
-   |         ^^^^^^^^
+   |         ^^^^^^^^ unreachable pattern
```

Comment:
Here I'd want to express "these patterns collectively make the last one unreachable" but I don't want to do a separate block with all the covering patterns: I like the readability of having the spans directly above the unreachable pattern span.

Reply:
Is what we do for cycle errors perhaps a useful model?  We highlight the bound that we can't check, and all of the steps of the cycle, and then say, "thus completing the cycle", etc.  This feels similar.

Reply:
I do see the similarity, how do you imagine this would look like in this case?

Reply:
Is this what you had in mind?
```
error: unreachable pattern
  --> $DIR/exhaustiveness-unreachable-pattern.rs:19:9
   |
LL |         (1 | 2,) => {}
   |         ^^^^^^^^ unreachable pattern
   |
note: these patterns collectively make the last one unreachable:
   |
LL |         (1,) => {}
   |         ---- matches some of the same values
LL |         (2,) => {}
   |         ---- matches some of the same values
LL |         (1 | 2,) => {}
   |         ^^^^^^^^ collectively making this unreachable
```
I like how it flows. I have a preference for not duplicating the main pattern if we can though.

Reply:
Probably, e.g.:

```
error: unreachable pattern
  --> $DIR/exhaustiveness-unreachable-pattern.rs:19:9
   |
LL |         (1 | 2,) => {}
   |         ^^^^^^^^ unreachable pattern
   |
note: these earlier patterns collectively make the later pattern unreachable:
   |
LL |         (1,) => {}
   |         ---- matches some of the same values
LL |         (2,) => {}
   |         ---- matches some of the same values
```

Reply:
I find it important for legibility that the covering patterns are shown before the unreachable one. I'd rather duplicate the unreachable one if we can't find a better phrasing

Reply:
Seems OK to me also to duplicate it.

Reply:
I did the one that duplicates, ty for the idea